images/singleuser-sample/Dockerfile

```
@@ -53,10 +48,16 @@ RUN apt-get update \
         git \
  && rm -rf /var/lib/apt/lists/*
 
-COPY --from=build-stage /build-stage/*.whl /tmp/pre-built-wheels/
+# set env for pip cache,
+# but use ARG to avoid persisting in image
+ARG PIP_CACHE_DIR=/tmp/pip-cache
+
+# install wheels built in the build-stage
 COPY requirements.txt /tmp/requirements.txt
-RUN pip install --no-cache-dir \
-        /tmp/pre-built-wheels/*.whl \
+RUN --mount=type=cache,target=${PIP_CACHE_DIR} \
+    --mount=type=cache,from=build-stage,source=/build-stage,target=/tmp/wheels \
+    pip install \
+        --find-links /tmp/wheels/ \
```

This variation helped my understanding a bit, not used `--find-links` before and wasn't sure for example that it accepted an argument or not.

```suggestion
    pip install --find-links=/tmp/wheels/ \
```